net-irc/anope/anope-2.0.7.ebuild

```diff
@@ -0,0 +1,155 @@
+			eerror "To upgrade, shut down Anope and copy your databases to"
+			eerror "${ROOT}/var/lib/anope and your configuration to ${ROOT}/etc/anope"
+			eerror "You can do this by:"
+			eerror "# mkdir -p ${ROOT}/var/lib/anope ${ROOT}/etc/anope"
+			eerror "# chown anope:anope ${ROOT}/var/lib/anope"
+			eerror "# mv ${ROOT}/opt/anope/data/*.db ${ROOT}/var/lib/anope"
+			eerror "# mv ${ROOT}/opt/anope/data/services.conf ${ROOT}/etc/anope"
+			die "Please move your anope database files from /opt/anope/data"
+		fi
+	done
+}
+
+src_prepare() {
+	anope_use_extra() {
+		local useflag=$1
+		local modulefile=$2
+		if use $useflag; then
+			ln -s "extra/${modulefile}" "modules/" || \
+				die "Failed to use ${modulefile}"
+		fi
+	}
+
+	anope_use_extra mysql			"m_mysql.cpp"
+	anope_use_extra sqlite			"m_sqlite.cpp"
+	anope_use_extra ssl			"m_ssl_openssl.cpp"
+	anope_use_extra gnutls			"m_ssl_gnutls.cpp"
+	anope_use_extra posix			"m_regex_posix.cpp"
+	anope_use_extra pcre			"m_regex_pcre.cpp"
+	anope_use_extra tre			"m_regex_tre.cpp"
+	anope_use_extra ldap			"m_ldap.cpp"
+	anope_use_extra sql-auth		"m_sql_authentication.cpp"
+	anope_use_extra sql-log			"m_sql_log.cpp"
+	anope_use_extra sql-oper		"m_sql_oper.cpp"
+	anope_use_extra ldap-auth		"m_ldap_authentication.cpp"
+	anope_use_extra ldap-oper		"m_ldap_oper.cpp"
+	anope_use_extra chanstats		"stats/m_chanstats.cpp"
+	anope_use_extra cs-fantasy-top		"stats/cs_fantasy_top.cpp"
+	anope_use_extra cs-fantasy-stats	"stats/cs_fantasy_stats.cpp"
+	anope_use_extra irc2sql			"stats/irc2sql/irc2sql.cpp"
+	anope_use_extra irc2sql			"stats/irc2sql/irc2sql.h"
+	anope_use_extra irc2sql			"stats/irc2sql/tables.cpp"
+	anope_use_extra irc2sql			"stats/irc2sql/utils.cpp"
+
+	# Avoid a silly sandbox error - tries to delete /usr/lib/modules
+	sed -i '/install.*REMOVE_RECURSE.*/d' CMakeLists.txt || die
+
+	cmake_src_prepare
+}
+
+src_configure() {
+	local mycmakeargs=(
+		-DBIN_DIR="libexec/anope"
+		-DDB_DIR="../var/lib/anope"
+		-DDOC_DIR="share/doc/${PF}"
+		-DLIB_DIR="$(get_libdir)/anope"
+		-DLOCALE_DIR="share/locale"
+		-DCONF_DIR="/etc/anope"
+		-DLOGS_DIR="../var/log/anope/"
+	)
+	cmake_src_configure
+}
+
+src_install() {
+	cmake_src_install
+	newinitd "${FILESDIR}/anope-init.d" "${PN}"
+	cp "${FILESDIR}/anope-conf.d-v2" "${T}" || die
+	sed -i "s~%LIBDIR%~${EPREFIX}/usr/$(get_libdir)/anope/~" \
+		 "${T}/anope-conf.d-v2" || die
+	newconfd "${T}/anope-conf.d-v2" "${PN}"
+	dosym ../libexec/anope/services /usr/bin/services
+	dosym ../libexec/anope/anopesmtp /usr/bin/anopesmtp
+	keepdir /var/log/anope /var/lib/anope/backups
+	fowners anope:anope /var/{lib,log}/anope /var/lib/anope/backups
+	dodoc -r docs/* data/example.conf
+	insinto /etc/anope
+	newins data/example.conf services.conf
+	fowners -R anope:anope /etc/anope
+	fperms -R 0700 /etc/anope
+	fperms 0755 /var/log/anope
+	fperms -R 0750 /var/lib/anope
+}
+
+pkg_preinst() {
+	if [[ -n ${REPLACING_VERSIONS} ]] ; then
```

Comment:
I am not sure if the ebuild should actually do something like that.
I get the idea but... imagine upgrade failed. You re-emerge package for some reason. Ebuild will now replace last good backup with a broken db. So you rely on that feature you will be in trouble now.

Reply:
Yeah, this is a really good point. It could fail during the compile or anywhere in the ebuild and harm the user.

I was nervous about touching the existing logic but I'm starting to see that being critical of everything is never a bad thing. This should just go now.